### ISSUE TITLE
WCAG 2.1 Accessibility Evaluation

### ISSUE THREAD

**Issue by Malvoz** [maplibre/maplibre-gl-js#53]
As a member of the W3C [Maps for HTML Community Group](https://www.w3.org/community/maps4html/) I set out to briefly evaluate the state of accessibility in popular web maps tools for the 2020 [W3C/OGC Joint Workshop Series on Maps for the Web](https://www.w3.org/2020/maps/) which included MapBox GL JS, thus the assessments largely apply to MapLibre as well. Some fixes were introduced in https://github.com/mapbox/mapbox-gl-js/pull/9991 which MapLibre inherited. 👍🏼 

The evaluation report lives here: https://github.com/Malvoz/web-maps-wcag-evaluation/blob/master/README.md

I'm hopeful that MapLibre will strive to build inclusive components for an accessible experience for its users. I suggest adding an "accessibility" label to issues relating to this topic, to encourage contributors to think about accessibility (and for issue tracking purposes of course).

Accessibility can be hard sometimes, consider adding a section about accessibility in the CONTRIBUTING.md document to advocate for building accessible content, perhaps including useful resources (something a long the lines of what I proposed for Leaflet in https://github.com/Leaflet/Leaflet/pull/7277).

MapLibre is a great initiative!

**Comment by HarelM** [comment]
@Malvoz was this issue covered by the other issues you opened?

**Comment by Malvoz** [comment]
@HarelM Yes mostly.

In relation to the [report](https://github.com/Malvoz/web-maps-wcag-evaluation/blob/master/README.md) (although most issues were originally identified for MapBox which MapLibre inherited):

- [x] Filed [#355](https://github.com/maplibre/maplibre-gl-js/issues/355) to address a failure of <span title="Success Criterion">SC</span> [4.1.2 Name, Role Value](https://www.w3.org/TR/WCAG21/#name-role-value) (the Name part)
- [x] Filed [#356](https://github.com/maplibre/maplibre-gl-js/issues/356) to address a failure of <span title="Success Criterion">SC</span> [4.1.2 Name, Role Value](https://www.w3.org/TR/WCAG21/#name-role-value) (the Role part)
- [x] Filed [#357](https://github.com/maplibre/maplibre-gl-js/issues/357) to address a failure of <span title="Success Criterion">SC</span> [4.1.2 Name, Role Value](https://www.w3.org/TR/WCAG21/#name-role-value) (the Value part)
- [x] Filed [#358](https://github.com/maplibre/maplibre-gl-js/issues/358) to address a failure of <span title="Success Criterion">SC</span> [1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG21/#info-and-relationships)
- [x] Filed [#359](https://github.com/maplibre/maplibre-gl-js/issues/359) to address an issue related to <span title="Success Criterion">SC</span> [1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG21/#info-and-relationships)
- [x] Filed [#360](https://github.com/maplibre/maplibre-gl-js/issues/360) to address a failure of <span title="Success Criterion">SC</span> [1.1.1 Non-Text Content](https://www.w3.org/TR/WCAG21/#non-text-content)
- [x] Filed [#361](https://github.com/maplibre/maplibre-gl-js/issues/361) to address a failure of <span title="Success Criterion">SC</span> [1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG21/#info-and-relationships) (due to a bug)
- [x] Filed [#362](https://github.com/maplibre/maplibre-gl-js/issues/362) to address an issue with double speak in some screen readers (e.g. the [NVDA bug](https://github.com/nvaccess/nvda/issues/7841))
- [x] Filed [#363](https://github.com/maplibre/maplibre-gl-js/issues/363) to address a failure of <span title="Success Criterion">SC</span> [2.5.5 Target Size](https://www.w3.org/TR/WCAG21/#target-size) failure
- [x] Filed [#364](https://github.com/maplibre/maplibre-gl-js/issues/364) to address a failure of <span title="Success Criterion">SC</span> [1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG21/#info-and-relationships) failure

There are certainly other issues, but I think this is a good start to improve the accessibility of MapLibre maps.

**Comment by Malvoz** [comment]
> Accessibility can be hard sometimes, consider adding a section about accessibility in the CONTRIBUTING.md document to advocate for building accessible content, perhaps including useful resources (something a long the lines of what I proposed for Leaflet in Leaflet/Leaflet#7277).

I can submit a PR later to add an "Accessibility" section (under [CONTRIBUTING.md#recommended-reading](https://github.com/maplibre/maplibre-gl-js/blob/main/CONTRIBUTING.md#recommended-reading)?).
